winch/codegen/src/isa/x64/masm.rs

```diff
@@ -337,14 +339,52 @@ impl Masm for MacroAssembler {
         self.asm.cmp(src.into(), dst, size);
         self.asm.setcc(kind, dst);
     }
+
+    fn clz(&mut self, src: Reg, dst: Reg, size: OperandSize) {
+        if self.flags.has_lzcnt() {
+            self.asm.lzcnt(src, dst, size);
+        } else {
+            let scratch = regs::scratch();
+
+            // Use the following approach:
+            // dst = size.num_bits() - bsr(src) - is_not_zero
+            //     = size.num.bits() + -bsr(src) - is_not_zero.
```

Comment:
How do you feel about using a higher level syntax to describe the formula?

I find English is too abstract and having a syntax that has multiple operations on the same line makes it easier to reason about if the formula is correct or has an error.

Reply:
I mean, are you okay with having
```
// dst = size.num_bits() - bsr(src) - is_not_zero
//     = size.num.bits() + -bsr(src) - is_not_zero.
```

in the comment as opposed to omitting it entirely or trying to describe it in English instead.

Reply:
Oh, I think the way you have it is perfect.